### ISSUE TITLE
remove_prefix_from_state_dict function does not work

### ISSUE THREAD

**Issue by PapaMadeleine2022** [OML-Team/open-metric-learning#344]
It seems that `remove_prefix_from_state_dict` function does not work. You can check it again.

**Comment by AlekseySh** [comment]
It works as we expect. There is a test on it, so you can understand what is going on inside: 
https://github.com/OML-Team/open-metric-learning/blob/165d63b93cd8da5d1fc52ae7b4a95bbd026fb164/tests/test_oml/test_models/test_utils.py#L4

**Comment by PapaMadeleine2022** [comment]
I think the function of `remove_prefix_from_state_dict ` is odd.
For example,
After I train my model, I load my weight by, The state_dict is as follows:
```
state_dict.keys(): odict_keys(['model.resnet.conv1.weight', 'model.resnet.bn1.weight', 'model.resnet.bn1.bias'...
```
The same prefix of all keys is `model.resnet.`, and the suffix is all different. But I want remove the prefix, so I can not use the  `remove_prefix_from_state_dict` to do want I want.

**Comment by AlekseySh** [comment]
why not? just call `remove_prefix_from_state_dict(state_dic, trial_key='conv1.weight')`
the function takes the guaranteed suffix as input because you don't know in advance what is your prefix

**Comment by AlekseySh** [comment]
I'll rename `trial_key` -> `guaranteed_suffix` to make it more clear

**Comment by PapaMadeleine2022** [comment]
That's cool~
